ooni.go

```diff
@@ -123,12 +91,18 @@ func (c *Context) Init() error {
 	return nil
 }
 
-// NewContext instance.
+// NewContext creates a new context instance.
```

Comment:
I see what you mean with the context name being a bit confusing.

Perhaps this can all end up inside of Session?

Reply:
I think not all of this can, as there are probe specific bits. I think we should continue this integration of probe-engine and, in the end, rename `Context` maybe `Probe`.

Importing `ooni` and using `ooni.Probe`would not surprise me for a tool called `ooniprobe`.

Actually, I do not see fields in `Context` that it makes sense moving to engine, but:

```Go
// Probe is a ooniprobe instance
type Probe struct {
	Config   *config.Config          // probe specific and `config.json` specific
	DB       sqlbuilder.Database   // definitely probe specific
	IsBatch  bool                           // probe specific: the engine doesn't have a real logger
	Session  *session.Session

	Home    string  // probe specific: but the engine uses a subdir
	TempDir string  // not sure what's the purpose of this :-)

	dbPath     string // also probe specific
	configPath string // ditto: I see the engine configured by calling code
}
```

the final name seems actually nice, no? (I've added comments explaining why I think there is no need to move more variables to the engine currently)

Reply:
Ah yes probe could also be a good name!

Reply:
See https://github.com/ooni/probe-cli/issues/45